### ISSUE TITLE
TypeError in convert_fats_to_dict() When fleet (name) Is NULL/None

### ISSUE THREAD

**Issue by Eingang** [ppfeufer/allianceauth-afat#115]
## TL;DR:
If `AFatLink.fleet` is `None` (because the corresponding DB field was `NULL`), `helper/views.helper.py` throws a `TypeError` trying to set the `fleet_name` value in the `summary` dictionary in `convert_fats_to_dict()`.

## System Information
- allianceauth==2.9.2
- allianceauth-afat==2.2.13
- allianceauth-imicusfat==1.4.2
- Django==3.2.9

## Discussion
Although the web-based form for adding an `AFatLink` says a fleet name is mandatory, the actual model says it can be null. However, `convert_fats_to_dict()` in `helper/views_helper.py` doesn't expect it to be null and will throw the following `TypeError` when it is null:

> `TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'` 

around line 212:

```python
       "fleet_name": fat.afatlink.fleet + esi_fleet_marker,
```

In my case, I migrated from earlier FAT modules and some FCs were terrible about filling in fleet names, so I have a lot of null `fleet` fields. Previous plug-ins displayed the hash as the fleet name if the fleet name was null, so it wasn't much of an issue. In AFAT, it was throwing server 500 errors to the AJAX character FAT loader when people loaded their personal fleet tracking dashboard.

## Suggested Resolutions
This can be addressed in several ways and it might be appropriate to employ more than one method depending on the approach chosen. Some suggestions:

1. When importing data from another FAT system, set null fleet names to some value, whether that's the hash for the fleet, the FC's name + fleet type + date, or some literal, e.g., "Unnamed Fleet".

1. Modify the `convert_fats_to_dict()` method to set a default value for fleet name if it's a `NoneType` value, even that's only an empty string.

1. Make null non-permissible in the `AFatLink` model for `fleet`.

1. In the instructions for importing from another system, some SQL preparation is suggested. Add an additional instruction to set the fleet names for existing NULL `fleet` field values to something, e.g. for BetterFat, 
     ```sql
      UPDATE bfat_fatlink SET `fleet` =  `hash` WHERE `fleet` IS NULL;
      ```

It might be appropriate to employ more than one method depending on the approach chosen.

## Conclusion
I solved my problem by setting `afat_afatlink.fleet` to `'Unnamed Fleet'` via a SQL UPDATE command, trusting that form handling logic for FATlink creation will appropriately avoid the issue in future by enforcing fleet name as a mandatory form field as the form claims.

**Comment by ppfeufer** [comment]
Hey, thanks for spotting this one. Didn't have an import from bFat in ages. I'll have a look at this.

**Comment by Eingang** [comment]
I had a quick look at the three import tasks when I was writing this up. It didn't look like any of them checked if the `fleet` field was empty before saving the record in your table. 

Coincidentally, as I finished writing this up and fixing mine, I flipped back to Discord to fine someone reporting this causing an AJAX error on their personal dashboard. Heh.

Thanks!

**Comment by ppfeufer** [comment]
v2.2.15 should fix it
